spec/var-op-test.test.sh

```diff
@@ -399,6 +399,76 @@ argv=minus
 argv=
 ## END
 
+#### $@ ("") and - and +
+set -- ""
+echo argv=${@-minus}
+echo argv=${@+plus}
+echo argv=${@:-minus}
+echo argv=${@:+plus}
+## STDOUT:
+argv=
+argv=plus
+argv=minus
+argv=
+## END
+
+# Zsh treats $@ as an array unlike Bash converting it to a string by joining it
+# with a space.
+
+## OK zsh STDOUT:
+argv=
+argv=plus
+argv=
+argv=plus
+## END
+
+#### $@ ("" "") and - and +
+set -- "" ""
+echo argv=${@-minus}
+echo argv=${@+plus}
+echo argv=${@:-minus}
+echo argv=${@:+plus}
+## STDOUT:
+argv=
+argv=plus
+argv=
+argv=plus
+## END
+
+#### $* ("" "") and - and + (IFS=)
+set -- "" ""
+IFS=
+echo argv=${*-minus}
+echo argv=${*+plus}
+echo argv=${*:-minus}
+echo argv=${*:+plus}
+## STDOUT:
+argv=
+argv=plus
+argv=
+argv=plus
+## END
+## BUG mksh/osh STDOUT:
```

Comment:
I usually let OSH fail and adjust allowed_failures

i.e. we don't usually have BUG osh, which is yellow - we make it RED 

This is basically a matter of convention

Reply:
I actually wondered if it would be possible to add another type of confirmed failure (e.g. "TODO") that produces red characters (or characters of another color) in the table. With the current strategy,

- When the number of allowed failures is increased by some changes, we cannot identify which failure was the increased one. We need to check out an older commit, build it, and compare the test results manually. Or one needs to open the github-jobs website, looking up the corresponding page, and compare the results.
- Even when the problem is fixed in the process of implementing another feature, the failure can be left to be FAIL if the result does not *exactly* match the expected result (which probably describes the Bash behavior).

Reply:
Hm well technically ALL the ~100 failures now are TODO !   :-)   Some of them will be classified as "OK osh", but never "BUG osh" I think ... our bugs are just things we haven't done yet

And notice there are ZERO FAIL on bash/dash/mksh/zsh ... so FAIL is really reserved for OSH

and BUG is reserved for the other shells

At least that is the convention I follow, but we can talk about it more if it causes a problem

Reply:
I'm not sure if I could call it "a problem", but at least the following situation can be improved:

> * When the number of allowed failures is increased by some changes, we cannot identify which failure was the increased one. We need to check out an older commit, build it, and compare the test results manually. Or one needs to open the github-jobs website, looking up the corresponding page, and compare the results.
> * Even when the problem is fixed in the process of implementing another feature, the failure can be left to be FAIL if the result does not _exactly_ match the expected result (which probably describes the Bash behavior).